### PR TITLE
C-Sky and MCore support for r2

### PR DESCRIPTION
**Pull request by wargio**
Example:
```
[0x08000034]> s sym.libat_add_fetch_1 
[0x08000060]> af
[0x08000060]> pdf
            ;-- __atomic_add_fetch_1:
╭ (fcn) sym.libat_add_fetch_1 44
│   sym.libat_add_fetch_1 ();
│           ; CALL XREFS from sym.libat_add_fetch_1 (0x800006e, 0x800007a)
│           0x08000060      f024           subi r0, 0xf
│           0x08000062      009d           st.w r13, (r0, 0x0)
│           0x08000064      109e           st.w r14, (r0, 0x4)
│           0x08000066      209f           st.w r15, (r0, 0x8)
│           0x08000068      2d12           mov r13, r2
│           0x0800006a      3e12           mov r14, r3
│           0x0800006c      4e01           zextb r14, r14
│           0x0800006e      077f           jsri [0x800008a]
│           0x08000070      0da3           ld.b r3, (r13, 0x0)
│           0x08000072      3e1c           addu r14, r3
│           0x08000074      4e01           zextb r14, r14
│           ; CALL XREF from sym.libat_add_fetch_1 (0x800006e)
│           0x08000076      0dbe           st.b r14, (r13, 0x0)
│           0x08000078      d212           mov r2, r13
│           0x0800007a      057f           jsri [0x800008e]
│           0x0800007c      e212           mov r2, r14
│           0x0800007e      208f           ld.w r15, (r0, 0x8)
│           ; CALL XREF from sym.libat_add_fetch_1 (0x800007a)
│           0x08000080      108e           ld.w r14, (r0, 0x4)
│           0x08000082      008d           ld.w r13, (r0, 0x0)
│           0x08000084      f020           addi r0, r0, 0xf
│           0x08000086      cf00           jmp r15
│       ╭─< 0x08000088      fff7           br 0x8000888
╰      ╭──< 0x0800008a      fff7           br 0x800088a
```
regressions/tests will be added later

### CODE REVIEW

**Comment by radare** [comment]
😍

**Comment by Maijin** [comment]
It's green now

**Comment by wargio** [comment]
wait till it gets green again.

File: libr/anal/p/anal_mcore.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
+/* radare - LGPL3 - Copyright 2018 - deroad */
+
+#include <r_anal.h>
+#include <r_types.h>
+#include <r_lib.h>
+#include "../../asm/arch/mcore/mcore.h"
+
+static int mcore_anal(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *buf, int len) {
+	mcore_handle handle = {0};
+    mcore_t* instr = NULL;
```
Comment (radare) [inline libr/anal/p/anal_mcore.c]: use tabs

File: libr/anal/p/anal_mcore.c [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+			break;
+        }
+        mcore_free (instr);
+    }
+	return op->size;
+}
+
+static int set_reg_profile(RAnal *anal) {
+	const char *p = 
+			"=PC	pc\n"
```
Comment (radare) [inline libr/anal/p/anal_mcore.c]: ugly indentation

File: libr/asm/p/asm_mcore.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    mcore_t* instr = NULL;
+	char tmp[256];
+    if (mcore_init(&handle, buf, len)) {
+        return -1;
+    }
+
+	if (op) {
+		memset (op, 0, sizeof (RAsmOp));
+		op->size = 2;
+	    if ((instr = mcore_next(&handle))) {
```
Comment (radare) [inline libr/asm/p/asm_mcore.c]: bad indent

Comment (radare) [reply libr/asm/p/asm_mcore.c]: space befor e(

**Comment by radare** [comment]
cc @xarkes @wargio

**Comment by wargio** [comment]
@radare up to you. these comes from the r2con issue list.

**Comment by radare** [comment]
Lets merge for now

**Pull request merged**